### PR TITLE
ddl: fix drop index failed when index contain auto_increment column and more than 2 index contain auto_increment_column (#12230) (#15861)

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve?

cherry-pick https://github.com/pingcap/tidb/pull/12230
https://github.com/pingcap/tidb/pull/15861

### CODE REVIEW

File: ddl/index.go [hunk 1]
```diff
@@ -15,6 +15,7 @@ package ddl
 
 import (
 	"context"
+	"github.com/pingcap/tidb/meta/autoid"
```
Comment (Deardrops) [inline ddl/index.go:18]: format

Comment (crazycs520) [reply ddl/index.go]: great, done.

**Pull request merged**